### DISCUSSION TITLE
Composite value

### DISCUSSION THREAD

**Discussion by gosarinamul** [WellyKorgis/project-blind-api#52] [Ideas]
## 👋 

`https://velog.io/@max9106/JPA-Value-%ED%83%80%EC%9E%85-%EB%A7%A4%ED%95%91` tag나 추후에 추가할 like 등등은 독립적인 테이블을 갖기에는 조금 부족하다고 생각이 들어서 이 composite value로 바꿀까 하는 데 어떠신가유..??

**Comment by nmsohn** [comment]
굉장히 좋은 제안이네요. 객체는 분리하되 데이터베이스 상에서는 같은 테이블에 넣고 싶을 경우 말씀하신대로 embed 하는 편이 좋을 것 같습니다. 

like의 경우 post 테이블에 넣는 것이 더 좋은 생각인 것 같아요. 굳굳👍

tag의 경우, 약간의 리서치를 해보았는데 [이곳](https://www.baeldung.com/jpa-tagging)처럼 string 배열로 처리하는 방법이 있을 수 있고 [여기](https://www.baeldung.com/jpa-tagging-advanced) 5번 예시처럼 eneity를 따로 두는 방법도 있고 1,2번처럼 endorsement 나 location 태그를 @embeddable하게 만들어 사용하는 경우가 있네요. 

tag를 클릭했을 시 관련 post를 전부 가져와 리스트 하는 기능이 있다고 가정했을 때 성능상 어떤 차이가 있는지 고민이 필요할 것 같습니다.  확실한 건 아니지만 tag array가 post 테이블에 존재할 때 쿼리하는 게 개발하면서 더 복잡성이 생기지 않을까요? 대략적으로 성능 비교를 해보려면 sql문으로 직접 비교해보는 게 빠를 것 같습니다.

결론적으로 사실 연습 프로젝트라 대량 데이터를 처리하는 일이 없기 때문에 tag를 post 테이블에 포함시켜도 크게 문제는 없다고 봐요😁😁😁

여담이지만 @embeddable 예시를 보니 합성키로 활용할 수 있네요. 이것도 써먹으면 좋을 것 같습니다.
